### PR TITLE
docs: add Conventions & Pitfalls master page

### LINKED ISSUE #3892: Conventions & Pitfalls master documentation page
One canonical page answering: tensor layout, (x,y) vs (h,w), align_corners defaults, rotation direction (2D screen vs 3D math convention), homography direction/normalization (warp_perspective vs homography_warp), bbox inclusive arithmetic, quaternion order, hue range. Addresses the recurring convention-confusion class: #2950, #777, #1142, #317. Follow-up to #3890 (llms.txt) — the page content also feeds llms-full.txt. Later stage: per-op convention blocks in geometry/augmentation docstrings.

### PR DESCRIPTION
**Pull request by ducha-aiki**
Fixes #3892

Adds `docs/source/get-started/conventions.rst` — the canonical "what convention does this function use" page (W1 of the agent-era plan), covering tensor layout, (x,y) vs (h,w) ordering, 2D-vs-3D rotation direction, `warp_perspective` vs `homography_warp` homography direction/normalization, per-function `align_corners` defaults, bbox inclusive arithmetic, quaternion WXYZ order, hue range, and a 15-item pitfall checklist. Every code block carries its own asserts and was executed against this checkout; the two llms files link to and extend from the page (write-once policy).

Addresses the convention-confusion issue class: #2950, #777, #1142, #317.

**Verification — empirical checks (kornia 0.9.0rc1, torch 2.9.1):**

```
bbox (1,1)-(2,2) h,w: 2.0 2.0                       (inclusive +1 confirmed, #1142)
homography_warp(dst->src normalized) == warp_perspective(src->dst pixel): True   (#2950)
axis_angle +90deg about z, R[:2,:2]: [[-0., -1.], [1., -0.]]   (math convention)
rotate(+90) marker (row1,col3) -> [[1, 1]]           (screen-CCW; opposite of axis_angle on y-down)
identity quaternion coeffs: [1.0, 0.0, 0.0, 0.0]     (WXYZ)
hue(red), hue(green): 0.0 2.094395                   (radians, 2pi/3)
meshgrid default range: -1.0 1.0
bbox_xyxy inverse round-trip max err: 0.0
```

**Verification — all 8 page code blocks executed (asserts inside are the tests):**

```
block 0..7: OK
```

**Verification — llms files and links:**

```
$ python3 .github/scripts/check_doc_links.py --offline
35 markdown file(s), 210 external link(s)  ...  no dead links (exit 0)
$ pixi run lint  -> ruff check/format Passed
$ pixi run build-docs -> get-started/conventions.html, llms.txt, llms-full.txt all in build root
```

The conventions.html URL inside the llms files will 404 until this merges and RTD rebuilds `latest` (same pattern as #3891).

Posted on behalf of @ducha-aiki by Claude (Fable 5).

### CODE REVIEW

File: docs/source/get-started/conventions.rst [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+    import torch
+    from kornia.geometry.conversions import normalize_homography
+    from kornia.geometry.transform import homography_warp, warp_perspective
+
+    img = torch.rand(1, 1, 8, 8)
+    M = torch.tensor([[[1.0, 0.0, 2.0], [0.0, 1.0, 1.0], [0.0, 0.0, 1.0]]])  # src->dst, pixels
+
+    a = warp_perspective(img, M, (8, 8))
+    M_norm_inv = normalize_homography(torch.inverse(M), (8, 8), (8, 8))
+    b = homography_warp(img, M_norm_inv, (8, 8), align_corners=True)
```
[the remark this answers is not in the document]
Comment (ducha-aiki) [reply docs/source/get-started/conventions.rst]: Confirmed and fixed in 5a0dd705. Reproduced the failure with unequal sizes: warping to (16, 8), the invert-first form with unswapped sizes diverges from `warp_perspective`, while normalize-then-invert matches (`torch.allclose` passes). The example now uses unequal src/dst sizes so its assert guards the general case, and the prose states the ordering explicitly.

*— reply written by Claude (Claude Code) on behalf of @ducha-aiki*

File: docs/source/_extra/llms-full.txt [hunk 2]
```diff
@@ -146,6 +148,12 @@ img = img[None]  # most ops want (B, C, H, W)
 6. Radians where degrees are expected — rotation angles are degrees.
 7. Uint8 `[0, 255]` tensors — ops expect float `[0, 1]`; divide by 255 first.
 8. Augmenting image and mask through two separate augmentation calls — the random parameters will differ; use one `AugmentationSequential` with `data_keys`.
+9. Feeding `homography_warp` a source→destination pixel homography — it expects destination→source, normalized to [-1, 1] (`normalize_homography(torch.inverse(M), size_src, size_dst)`), and defaults to `align_corners=False` unlike `warp_perspective`.
```
[the remark this answers is not in the document]
Comment (ducha-aiki) [reply docs/source/_extra/llms-full.txt]: Fixed in 5a0dd705 — the pitfall now reads `torch.inverse(normalize_homography(M, size_src, size_dst))` (normalize the forward homography first, then invert).

*— reply written by Claude (Claude Code) on behalf of @ducha-aiki*

**Comment by ducha-aiki** [comment]
Qodo findings addressed:

1. **Wrong homography size order** — confirmed and fixed in 5a0dd705 using the normalize-then-invert form (`torch.inverse(normalize_homography(M, size_src, size_dst))`); the page example now warps to unequal src/dst sizes so its assert guards the general case. Same recipe fixed in `llms-full.txt`.
2. **Pitfall lists out of sync** — fixed in e27c2e96, but by deduplication rather than propagation: the page's item 14 restated item 4 (both said the `align_corners` defaults differ), so it was removed. Page and `llms-full.txt` now carry the same 14 items in the same order.
3. **Angle convention contradiction** — confirmed and fixed in e27c2e96: both llms files now state the split the page documents (degrees in the 2D image APIs, radians in the 3D/conversion APIs) instead of the blanket "degrees" claim.

All page code blocks re-executed clean after the changes; offline link check passes.

*— comment written by Claude (Claude Code) on behalf of @ducha-aiki*
